📝(project) refacto documentation for Ralph V4 (#492)
openfun/ralph: docs/tutorials/start-dev-with-ralph.md

Reviewer: I would put this tutorial under a `Development - Contributing` section, WDYT?

> # How to start developing with Ralph ?

Reviewer: ```suggestion
# How to start developing with Ralph?
```

### Prepare your working environment 

To start developing with Ralph, you must install the following dependencies on your local environment: 

- **Python 3.11**
- make
- Docker/ Docker compose

### Bootstrap Ralph for development

> To start playing with `ralph`, you should build it using the `bootstrap` Make target:

Reviewer: ```suggestion
To start playing with `ralph`, you should first bootstrap the project:
```

```bash
make bootstrap
```

Once the project has been bootstrapped, you may want to edit generated `.env`

> file to set up available backends parameters that will be injected in the

Reviewer: ```suggestion
file to set up available backends parameters that will be injected into the
```

> Once the project has been bootstrapped, you may want to edit generated `.env`
> file to set up available backends parameters that will be injected in the

Reviewer 2: ```suggestion
Once the project has been bootstrapped, you may want to edit the generated `.env`
file to set up available backend parameters that will be injected in the
```

running container as environment variables to configure Ralph (see [backends
documentation](./backends.md)):

```bash
# Elasticsearch backend
RALPH_ES_HOSTS=http://elasticsearch:9200
RALPH_ES_INDEX=statements
RALPH_ES_TEST_HOSTS=http://elasticsearch:9200
RALPH_ES_TEST_INDEX=test-index

# [...]
```

> Note that lines starting with a `#` are considered as commented and thus will
> have no effect while running Ralph.

### Working with backends

To configure the backends, we provide default parameters in the `.env.dist`

> template, you can copy/paste them in your `.env` file (and uncomment them so
> that they are properly injected in running containers).

Reviewer: ```suggestion
template which is copied to the `.env` file when bootstrapping the project for the first time. You can uncomment them so that they are properly injected in running containers.
```

> In order to run the Elasticsearch backend locally on GNU/Linux operating
> systems, ensure that your virtual memory limits are not too low and increase

> > them (temporally) if needed by typing this command from your terminal (as

Reviewer: ```suggestion
> them if needed by typing this command from your terminal (as
```
Not sure about the signification of "temporally" here

> `root` or using `sudo`):
>
> `sysctl -w vm.max_map_count=262144`
>
> Reference:
> https://www.elastic.co/guide/en/elasticsearch/reference/master/vm-max-map-count.html

> 

Reviewer 2: Perhaps we should also mention that we need at least 10% of remaining disk space to run Elasticsearch ?

> Once configured, start backends using:

Reviewer: ```suggestion
Once configured, start the backend container using:
```

Once your Kubernetes cluster is running, you need to create a standard Arnold
project describing applications and environments you need to deploy:

```bash
make arnold-bootstrap
```

Once bootstrapped, Arnold should have created a `group_vars` directory
containing the following files:

```bash
tree group_vars
group_vars
├── common
└── customer
    └── ralph
        ├── development
        │   ├── main.yml
        │   └── secrets
        │       ├── databases.vault.yml
        │       ├── elasticsearch.vault.yml
        │       └── ralph.vault.yml
        └── main.yml

5 directories, 5 files
```

To create the LRS credentials file, you need to provide a list of accounts
allowed to request the LRS in Ralph's vault:

```bash
# Setup your kubernetes environment
source .k3d-cluster.env.sh

# Decrypt the vault
bin/arnold -d -c ralph -e development -- vault -a ralph decrypt
```

Edit the vault file to add a new account for the `foo` user with the `bar`
password and a relevant scope:

```yaml
# group_vars/customer/ralph/development/secrets/ralph.vault.yml
#
# [...]
#
# LRS
LRS_AUTH:
  - username: "foo"
    hash: "$2b$12$lCggI749U6TrzK7Qyr7xGe1KVSAXdPjtkMew.BD6lzIk//T5YSb72"
    scopes:

>       - "foo_scope"

Reviewer 2: ```suggestion
      - "all"
```
We should provide valid scopes in the examples.

`all` would be an admin user

xAPI spec specifies `The LRS MUST assume a requested scope of "statements/write" and "statements/read/mine" if no scope is specified.`, which could be an alternative default value.
